plugins/gin-gonic/gin/doc.go

```diff
@@ -0,0 +1,22 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
I think the gin stuff mistakenly was included in this PR as well.

Reply:
The gin PR adds the internal/trace code this PR requires, so this PR is made on top of the gin PR. But since gin PR is blocked on legal stuff, I'm going to extract the internal/trace code from gin PR and put it here to unblock this PR. It should be fine that way, because none of the code in internal/trace is based on the datadog code, and none of the code in this PR is based on datadog code, so all should be fine from legal point of view.